### ISSUE TITLE
morphedByMany and composite primary key not working

### ISSUE THREAD

**Issue by merlosy** [vuex-orm/vuex-orm#400]
Hi,
I have a issue with polymorphic many-to-many relationship using a composite primary key.
It's related to https://vuex-orm.github.io/vuex-orm/guide/model/relationships.html#defining-the-inverse-of-the-relationship-2

But, it's like the `posts` entity has a composite primary key.

Basically it returns a list of items, but they are all `undefined`.

My pivot table is like this (just not called taggable):
```ts
export class CollectionnableEntity extends Model {
    static entity = 'app-collectionnable';

    static primaryKey = [
        'collection_id',
        'collectionnable_id',
        'collectionnable_type'
    ];

    static fields() {
        return {
            collection_id: this.attr(null),
            collectionnable_id: this.attr(null),
            collectionnable_type: this.attr(null)
        };
    }
}
```

**Comment by merlosy** [comment]
It could be nice to be able to specify the `separator` (static string) between fields for composite primary keys. This would probably ease splitting things for pivot tables, or use a different separator (like double underscore `__`) for joining keys in pivot tables.

**Comment by kiaking** [comment]
@merlosy Hi! Thank you for the report! Hm... I haven't tested with 3 campsite key in fact... I should write test case on this.

> It could be nice to be able to specify the separator (static string) between fields for composite primary keys. This would probably ease splitting things for pivot tables, or use a different separator (like double underscore __) for joining keys in pivot tables.

Yeah, current composite key handling is not ideal. We had another idea on storing array as an string like `'[1, 1, 2]'` and extract them when we need to use it. We should continue discuss on how to handle I think.

**Comment by kiaking** [comment]
@merlosy Hi! It has been a while, but do you have any reproducible code on this one?

**Comment by kiaking** [comment]
Closing the issue due to inactivity. Please feel free to reopen it if there is anything! 🤝

**Comment by merlosy** [comment]
why was this closed?

**Comment by Tofandel** [comment]
@kiaking Can we reopen, it would be good to fix this bug I can have a look

**Comment by cuebit** [comment]
@Tofandel you can create an issue in the [next](https://github.com/vuex-orm/vuex-orm-next/issues) repo since this repo will no longer receive updates as per the [README](https://github.com/vuex-orm/vuex-orm/blob/master/README.md).

**Comment by Tofandel** [comment]
Maybe we can get a backward fix for 0.36 as well when fixed on next as well as this is a bug?

**Comment by cuebit** [comment]
@Tofandel In that case your best option would be to fork the repo and patch the lib with a fix for your own use. Remedial work such as this is far too extensive for a version that is no longer maintained and has a successor in progress. Sorry!
